### PR TITLE
[Merged by Bors] - feat: port Data.PNat.Basic

### PR DESCRIPTION
**Pull request by joneugster**
Fairly straight forward, with no bigger issues.

### CODE REVIEW

File: Mathlib/Data/PNat/Basic.lean [hunk 1]
```diff
@@ -19,95 +18,98 @@ import Mathlib.Algebra.Order.Positive.Ring
...
 
 namespace PNat
 
 @[simp]
-theorem one_add_nat_pred (n : ℕ+) : 1 + n.natPred = n := by
-  rw [nat_pred, add_tsub_cancel_iff_le.mpr <| show 1 ≤ (n : ℕ) from n.2]
-#align pnat.one_add_nat_pred PNat.one_add_nat_pred
+theorem one_add_natPred (n : ℕ+) : 1 + n.natPred = n := by
+  rw [natPred, add_tsub_cancel_iff_le.mpr <| show 1 ≤ (n : ℕ) from n.2]
+#align pnat.one_add_natPred PNat.one_add_natPred
```
Comment (jcommelin) [inline Mathlib/Data/PNat/Basic.lean:34]: This looks like a some search-replace went wrong

Comment (joneugster) [reply Mathlib/Data/PNat/Basic.lean]: oops O.o

File: Mathlib/Data/PNat/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,458 @@
...
+
+@[simp]
+theorem natPred_add_one (n : ℕ+) : n.natPred + 1 = n :=
+  (add_comm _ _).trans n.one_add_natPred
+#align pnat.nat_pred_add_one PNat.natPred_add_one
+
+-- Porting note: not implemented yet.
+-- @[mono]
+theorem natPred_strictMono : StrictMono natPred := fun m _ h => Nat.pred_lt_pred m.2.ne' h
+#align pnat.nat_pred_strictMono PNat.natPred_strictMono
```
Comment (jcommelin) [inline Mathlib/Data/PNat/Basic.lean:44]:
```suggestion
#align pnat.nat_pred_strict_mono PNat.natPred_strictMono
```

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Data/PNat/Basic.lean [hunk 3]
```diff
@@ -0,0 +1,458 @@
...
+
+theorem mod_div_aux_spec :
+    ∀ (k : ℕ+) (r q : ℕ) (_ : ¬(r = 0 ∧ q = 0)),
+      ((modDivAux k r q).1 : ℕ) + k * (modDivAux k r q).2 = r + k * q
+  | k, 0, 0, h => (h ⟨rfl, rfl⟩).elim
+  | k, 0, q + 1, _ => by
+    change (k : ℕ) + (k : ℕ) * (q + 1).pred = 0 + (k : ℕ) * (q + 1)
+    rw [Nat.pred_succ, Nat.mul_succ, zero_add, add_comm]
+  | k, r + 1, q, _ => rfl
+#align pnat.mod_div_aux_spec PNat.mod_div_aux_spec
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/PNat/Basic.lean:356]:
```suggestion
theorem modDivAux_spec :
    ∀ (k : ℕ+) (r q : ℕ) (_ : ¬(r = 0 ∧ q = 0)),
      ((modDivAux k r q).1 : ℕ) + k * (modDivAux k r q).2 = r + k * q
  | k, 0, 0, h => (h ⟨rfl, rfl⟩).elim
  | k, 0, q + 1, _ => by
    change (k : ℕ) + (k : ℕ) * (q + 1).pred = 0 + (k : ℕ) * (q + 1)
    rw [Nat.pred_succ, Nat.mul_succ, zero_add, add_comm]
  | k, r + 1, q, _ => rfl
#align pnat.mod_div_aux_spec PNat.modDivAux_spec
```

File: Mathlib/Data/PNat/Basic.lean [hunk 4]
```diff
@@ -0,0 +1,458 @@
...
+
+theorem mod_add_div (m k : ℕ+) : (mod m k + k * div m k : ℕ) = m := by
+  let h₀ := Nat.mod_add_div (m : ℕ) (k : ℕ)
+  have : ¬((m : ℕ) % (k : ℕ) = 0 ∧ (m : ℕ) / (k : ℕ) = 0) := by
+    rintro ⟨hr, hq⟩
+    rw [hr, hq, mul_zero, zero_add] at h₀
+    exact (m.ne_zero h₀.symm).elim
+  have := mod_div_aux_spec k ((m : ℕ) % (k : ℕ)) ((m : ℕ) / (k : ℕ)) this
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/PNat/Basic.lean:364]:
```suggestion
  have := modDivAux_spec k ((m : ℕ) % (k : ℕ)) ((m : ℕ) / (k : ℕ)) this
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Pull request closed without merging**